src/main/scala/scenarios/ProcessMintingRequest.scala

```diff
@@ -143,27 +159,27 @@ trait Minter {
                      else "NOT Running in dry mode - will attempt to process minting request and query node"
         println(dryMsg)
 
-        mintRequests.foreach{
-         case (message, mintRequest) =>
-           // ToDo handle failures here so that a single request failure does not taint the entire batch of sqs messages
-           println(s"mint request: $mintRequest")
-           if (!config.dry){
-              println(s"Attempting to process mint request box ${mintRequest.mintRequestBoxId} issued by mint tx ${mintRequest.mintTxId}")
-              // ToDo avoid creating an ergo client per message
-              // TODO: Update processMintingRequest to take Address instead of String
-              val txJson = processMintingRequest(
-                ergoNodeConfig,
-                mintingContractAddress,
-                mintRequest.mintRequestBoxId,
-                ergoNodeConfig.getParameters.get("ergoNamesTokenDescription"))
-
-              println("Successfully submitted minting tx to node")
-              println(txJson)
-           }
-
-           sqsClient.deleteMessage(queueUrl, message.getReceiptHandle)
-           print("Deleted message")
-       }
+        // looping throug the requests and try to mint them
+        val results = mintRequests.map{
+         case (message: SQSEvent.SQSMessage, mintRequest: MintRequestSqsMessage) =>
+            println(s"mint request: $mintRequest")
+            val r = Try(processSqsMessage(sqsClient, message, mintRequest, ergoClient, ergoNodeConfig, mintingContractAddress, config.dry, queueUrl))
+            (mintRequest, r)
+        }
+        val total = results.size
+        val successful = results.map(_._2.isSuccess).size
```

Comment:
@programmer1718  Thinking we might use this  list of unsucess mintings in the future.
i.e: figure out how many times a message have been re-tried and do a refund if need be?